CONTRIBUTING.md

```diff
@@ -1,25 +1,67 @@
 # Developer Guide
 
-# TODO (andreyvelich): This doc needs to be updated for Kubeflow Trainer V2
-
-Kubeflow Training Operator is currently at v1.
+This guide explains how to contribute to the Kubeflow Trainer V2 project.
 
 ## Requirements
 
 - [Go](https://golang.org/) (1.23 or later)
-- [Docker](https://docs.docker.com/) (23 or later)
+- [Docker](https://www.docker.com/) (20.10.0 or later)
 - [Python](https://www.python.org/) (3.11 or later)
 - [kustomize](https://kustomize.io/) (4.0.5 or later)
-- [Kind](https://kind.sigs.k8s.io/) (0.22.0 or later)
+- [Kind](https://kind.sigs.k8s.io/) (0.27.0 or later)
 - [Lima](https://github.com/lima-vm/lima?tab=readme-ov-file#adopters) (an alternative to DockerDesktop) (0.21.0 or later)
   - [Colima](https://github.com/abiosoft/colima) (Lima specifically for MacOS) (0.6.8 or later)
 - [pre-commit](https://pre-commit.com/)
 
 Note for Lima the link is to the Adopters, which supports several different container environments.
 
-## Building the operator
+## Development
+
+### Development Workflow
+
+The Kubeflow Trainer project includes a Makefile with several helpful commands to streamline your development workflow:
+
+```sh
+# Generate manifests, APIs and SDK
+make generate
+
+# Download required tools for development
+make ginkgo envtest controller-gen kind
+
+# Format and verify code
+make fmt vet golangci-lint
+
+# Run tests (see Testing section below)
+make test test-integration test-e2e
+
+# Setup a cluster for e2e testing
+make test-e2e-setup-cluster
```

Comment:
Let's move these 2 command to Testing, since they are not required for Development.
```suggestion
```